### PR TITLE
vast driver: random and roundrobin strategies for vast driver

### PR DESCRIPTION
**Pull request by vlboiko**

### CODE REVIEW

File: manila/share/drivers/vastdata/__init__.py [hunk 1]
```diff
@@ -78,6 +89,10 @@ def do_setup(self, context):
         self._host = self.configuration.vast_mgmt_host
         self._vippool_name = self.configuration.vast_vippool_name
         self._root_export = self.configuration.vast_root_export
+        try:
+            self._load_balancing = VipPoolManagementStrategy[str(self.configuration.vast_load_balancing).upper()]
+        except KeyError:
```
Comment (koreno) [inline manila/share/drivers/vastdata/__init__.py:94]: we should log a warning when the user input in the configuration is invalid (`"%s is invalid - falling back on 'RANDOM' VIP-Pool load-balancing strategy"`)

File: manila/share/drivers/vastdata/__init__.py [hunk 2]
```diff
@@ -221,14 +242,13 @@ def ensure_share(self, context, share, share_server=None):
         elif not export.policy == share_id:
             self.vms_session.patch("views/{}".format(export.id), data=dict(policy_id=policy.id))
 
-        return [dict(
+        return dict(
```
Comment (koreno) [inline manila/share/drivers/vastdata/__init__.py]: why? it used to return a `List[Dict]` type, how can we change it to a `Dict` now? Am I missing something?

(`ensure_share` isn't only used by `create_share`)

**Review by koreno — CHANGES_REQUESTED** [review]

File: manila/share/drivers/vastdata/__init__.py (hunk 1 above)
Comment (vlboiko) [reply manila/share/drivers/vastdata/__init__.py:94]: Done

File: manila/share/drivers/vastdata/__init__.py (hunk 2 above)
Comment (vlboiko) [reply manila/share/drivers/vastdata/__init__.py]: Yes, it is strange for me. List of dict was generaged before only for reason to take first dict in list on line 250. And it aways returned first pool before with the same ip.

Comment (koreno) [reply manila/share/drivers/vastdata/__init__.py]: if you `git grep` the manila project you'll find that the `ensure_share` is called from another location, where a list is expected. 
this is not the only place it is used.

**Pull request merged**